src/features/swap/components/SwapTxInfo/SwapTx.tsx

```diff
@@ -1,65 +1,24 @@
 import type { Order } from '@safe-global/safe-gateway-typescript-sdk'
 import type { ReactElement } from 'react'
 import { Box, Typography } from '@mui/material'
-import TokenIcon from '@/components/common/TokenIcon'
-import { formatVisualAmount } from '@/utils/formatters'
+import TokenAmount from '@/components/common/TokenAmount'
 
 export const SwapTx = ({ info }: { info: Order }): ReactElement => {
   const { kind, sellToken, sellAmount, buyToken, buyAmount } = info
   const isSellOrder = kind === 'sell'
 
-  let from = (
-    <>
-      <Box style={{ paddingRight: 5, display: 'inline-block' }}>
-        <TokenIcon logoUri={sellToken.logoUri || undefined} tokenSymbol={sellToken.symbol} />
-      </Box>
-      <Typography component="span" fontWeight="bold">
-        {formatVisualAmount(sellAmount, sellToken.decimals)} {sellToken.symbol}{' '}
-      </Typography>
-    </>
-  )
-
-  let to = (
-    <>
-      <Box style={{ paddingLeft: 5, paddingRight: 5, display: 'inline-block' }}>
-        <TokenIcon logoUri={buyToken.logoUri || undefined} tokenSymbol={buyToken.symbol} />
-      </Box>{' '}
-      <Typography component="span" fontWeight="bold">
-        {buyToken.symbol}
-      </Typography>
-    </>
-  )
+  let from = <TokenAmount value={sellAmount} decimals={sellToken.decimals} logoUri={sellToken.logoUri ?? undefined} />
+  let to = <TokenAmount value={buyAmount} decimals={buyToken.decimals} logoUri={buyToken.logoUri ?? undefined} />
 
   if (!isSellOrder) {
-    from = (
-      <>
-        <Box style={{ paddingRight: 5, display: 'inline-block' }}>
-          <TokenIcon logoUri={sellToken.logoUri || undefined} tokenSymbol={sellToken.symbol} />
-        </Box>
-        <Typography component="span" fontWeight="bold">
-          {sellToken.symbol}
-        </Typography>
-      </>
-    )
-    to = (
-      <>
-        <Box style={{ paddingLeft: 5, paddingRight: 5, display: 'inline-block' }}>
-          <TokenIcon logoUri={buyToken.logoUri || undefined} tokenSymbol={buyToken.symbol} />
-        </Box>{' '}
-        <Typography component="span" fontWeight="bold">
-          {formatVisualAmount(buyAmount, buyToken.decimals)} {buyToken.symbol}{' '}
-        </Typography>
-      </>
-    )
+    ;[from, to] = [to, from]
```

Comment:
@compojoom I remember we tried this in the first iteration of swaps but had to change it. Do you remember what the issue was?